chainerx_cc/chainerx/routines/arithmetic.cc

```diff
@@ -491,4 +491,34 @@ Array Power(const Array& x1, Scalar x2) { return internal::Binary(&PowerASImpl,
 
 Array Power(Scalar x1, const Array& x2) { return internal::Binary(&PowerSAImpl, x1, x2, GetArithmeticResultDtype(x1, x2)); }
 
+Array Fmod(const Array& x1, const Array& x2) {
+    CheckEqual(x1.shape(), x2.shape());
+    Array out = Empty(x1.shape(), x1.dtype(), x1.device());
+
+    {
+        NoBackpropModeScope scope{};
+        x1.device().backend().CallKernel<FmodKernel>(x1, x2, out);
+    }
+
+    {
+        BackwardBuilder bb{"fmod", x1, x2};
```

Comment:
```
tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py::test_Fmod_param_0_{in_dtypes=('int8', 'int8'), out_dtype='int8', in_shapes=((2, 3), (2, 3)), input_lhs='random', input_rhs='random', is_module=False}[native:0] call kernel
build backward
python3.7: ~/dev/chainer/chainerx_cc/chainerx/backward_builder.h:149: chainerx::BackwardBuilder::Target chainerx::BackwardBuilder::CreateTarget(std::vector<long unsigned int>): Assertion `input_index < inputs_target_created_.size()' failed.
```

Reply:
FYI: I'm running `pytest ./tests/chainerx_tests/unit_tests/routines_tests/test_arithmetic.py -k Fmod -v -s` after `pip install`

Reply:
After fixing as suggested test is still failing so please fix it!

Reply:
Have fixed the errors. Thanks for the suggestions.